component/src/main/java/org/ballerinalang/net/kafka/nativeimpl/KafkaBalServerConnector.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.net.kafka.nativeimpl;
+
+import org.ballerinalang.annotation.JavaSPIService;
+import org.ballerinalang.connector.api.Annotation;
+import org.ballerinalang.connector.api.BallerinaConnectorException;
+import org.ballerinalang.connector.api.BallerinaServerConnector;
+import org.ballerinalang.connector.api.Service;
+import org.ballerinalang.net.kafka.Constants;
+import org.ballerinalang.net.kafka.KafkaUtils;
+import org.ballerinalang.net.kafka.api.KafkaListener;
+import org.ballerinalang.net.kafka.api.KafkaServerConnector;
+import org.ballerinalang.net.kafka.exception.KafkaConnectorException;
+import org.ballerinalang.net.kafka.impl.KafkaListenerImpl;
+import org.ballerinalang.net.kafka.impl.KafkaServerConnectorImpl;
+import org.ballerinalang.util.exceptions.BallerinaException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+/**
+ * {@code KafkaBalServerConnector} This is the Kafka implementation for the {@code BallerinaServerConnector} API.
+ */
+@JavaSPIService("org.ballerinalang.connector.api.BallerinaServerConnector")
+public class KafkaBalServerConnector implements BallerinaServerConnector {
+
+    private static final Logger logger = LoggerFactory.getLogger(KafkaBalServerConnector.class);
+
+    private Map<String, KafkaServerConnector> connectorMap = new HashMap<>();
```

Comment:
Do we need this map? The values in the map are never used

Reply:
+1